### ISSUE TITLE
Generalise riboviz.test.regression.test_vignette

### ISSUE THREAD

**Issue by mikej888** [riboviz/riboviz#155]
Generalise to take any configuration file, optionally run the workflow, then validate the actual outputs (as held in the directories specified within the configuration file) against a directory of expected outputs.

Will be of help in #32.

**Comment by mikej888** [comment]
@FlicAnderson

`riboviz.test.regression.test_vignette` has been generalised and renamed to `test_regression`. It's in the [generalise-regression-test-vignette-155](https://github.com/riboviz/riboviz/tree/generalise-regression-test-vignette-155) branch.

It now takes a `--config-file` parameter with a configuration file which is used to find the files to be validated (using `dir_tmp` and `dir_out`). The `--expected` parameter is still used to specify the regression test data location. 

As an example of use, to validate output produced by a run with `vignette/simdata_umi_config.yaml` where output from a previous run is in `$HOME/simdata-umi-data/` (which contains `index`, `simdata_umi_tmp` and `simdata_umi_output` subdirectories), run:

```console
$ pytest -s riboviz/test/regression/test_regression.py --expected=$HOME/simdata-umi-data --config-file=vignette/simdata_umi_config.yaml --skip-workflow
```

To validate output produced by run using `vignette/vignette-config.yaml`, run:

```console
$ pytest -s riboviz/test/regression/test_regression.py --expected=$HOME/vignette-20200304-2.0.beta --skip-workflow
```

`docs/developer/testing.md` has updated usage and documentation.

Please let me know how you get on and if there are any issues.

**Comment by kavousan** [comment]
@FlicAnderson to test this.

**Comment by mikej888** [comment]
See pull request #155.

**Comment by FlicAnderson** [comment]
I still need to test this, haven't yet.

**Comment by FlicAnderson** [comment]
Still haven't tested this yet, will do before end of week starting 01st June, as part of submitting new pull request for merging #32 `regen-stats-32` code into develop.

**Comment by FlicAnderson** [comment]
Tested this, comparing 1 sample full-size data from G-Sc_2014 run in `develop` vs `regen-stats-32`.

Ran it between those file sets ("21 passed, 27 skipped in 71.65 seconds")  

I then manually edited one of the numbers in `read_counts.tsv` in `develop` file bundle and re-ran again with the same command, getting "**1 failed**, 20 passed, 27 skipped in 69.44 seconds" and errors pointing to the read_counts.tsv file in develop bundle. Exactly what we expected.

VERY NICE!  This will make my life so much easier!

In summary, `generalise-regression-test-vignette-155` branch: 
- runs vignette fine
- regression tests pass, matching `regression-test-data-20200221`
- merged latest develop into branch (commit 95178e2ef from 27th May), no conflicts.
- pytest all-tests all pass: 166 in 187.42 seconds
- coverage tests all passing, with 69% coverage
- documentation good once I'd gotten my file bundle locations sorted out, but as I use it more often, I'll see if I can figure out any specific points I can improve or if I just had a slightly slower brain day today :)

Happy to close issue! (and subsequent pull request)
